### DISCUSSION TITLE
Interactive tutorials for next.js

### DISCUSSION THREAD

**Discussion by vasucp1207** [vercel/next.js#45426] [Ideas]
Hey,
I have an idea for an interactive tutorial website for Next.js. By creating a step-by-step guide for learning Next.js, we can provide an engaging and hands-on experience for users to learn this popular framework like all new frameworks ex svelte, lit, and react(beta) using such interactive tutorials for better learning resources.

I currently also working on a similar project for [**jotai**(state-management library)](https://github.com/jotaijs/jotai-tutorial).

I believe this project will help us tap into the growing demand for Next.js skills and provide a valuable resource for those interested in web development.

Let's discuss further and see if we can bring this idea to life.

**Comment by guitheengineer** [comment]
[There's already a tutorial](https://nextjs.org/learn/foundations/about-nextjs?utm_source=next-site&utm_medium=nav-cta&utm_campaign=next-website), but I agree something more interactive can be done on the same way you did (e.g [svelte tutorial](https://svelte.dev/tutorial/basics), [solid](https://www.solidjs.com/tutorial/introduction_basics)) although I don't think it will be simple to have a server running as it's to just run JS.

**Reply by vasucp1207** [reply]
Creating an interactive tutorial can be a challenging task, especially if it requires a server to run. However, we can use code sandbox to simplify the process and it would be great if we can make the tutorial you mention above interactive.
